### ISSUE TITLE
Listen to new tweets using lists

### ISSUE THREAD

**Issue by bitog** [mahrtayyab/tweety#152]
Hi,

I have seen here in the issues that if you want to listen to new tweets its best to do that through a list.

I have been doing that, and it works out well, however I am not able to grab replies reliably. Anybody got an idea how to solve that issue?

Thank you.

**Comment by mahrtayyab** [comment]
as far as I know , replies aren't part of list timeline.

**Comment by harleyLuke** [comment]
This is my solution, if encountered Conversition in the list, I'll get Twitter Detail , and I add new field 'self_vertical_conversation' in Tweet to get he self  reply
https://github.com/harleyLuke/tweety/commit/1f7d065967a91f92fc1f60fa044bfa100c6dfbe3

and extra code

`    def convert_tw_list2dbjson(self,tw_list,list_id,cursor):
        for item in tw_list:
            if isinstance(item,Tweet):
                self._convert_tw2db_json(item,list_id,cursor)
            elif isinstance(item,SelfThread):
                self._convert_selfthread2db_json(item,list_id,cursor)
            else:
                print("get insert error")
                
`    def _convert_selfthread2db_json(self, tw_thread, list_id, cursor):

            tw_detail = self.get_twitter_by_twid(tw_id)
            self._convert_tw2db_json(tw_detail, list_id, cursor)
            for items in tw_detail.self_vertical_conversation:
                for item in items.tweets:
                    self._convert_tw2db_json(item, list_id, cursor)`                `

**Comment by mahrtayyab** [comment]
> This is my solution, if encountered Conversition in the list, I'll get Twitter Detail , and I add new field 'self_vertical_conversation' in Tweet to get he self reply [harleyLuke@1f7d065](https://github.com/harleyLuke/tweety/commit/1f7d065967a91f92fc1f60fa044bfa100c6dfbe3)
> 
> and extra code
> 
> `def convert_tw_list2dbjson(self,tw_list,list_id,cursor):` ` for item in tw_list:` ` if isinstance(item,Tweet):` ` self._convert_tw2db_json(item,list_id,cursor)` ` elif isinstance(item,SelfThread):` ` self._convert_selfthread2db_json(item,list_id,cursor)` ` else:` ` print("get insert error")`
> 
> `def _convert_selfthread2db_json(self, tw_thread, list_id, cursor):` ` tw_detail = self.get_twitter_by_twid(tw_id)` ` self._convert_tw2db_json(tw_detail, list_id, cursor)` ` for items in tw_detail.self_vertical_conversation:` ` for item in items.tweets:` ` self._convert_tw2db_json(item, list_id, cursor)`

How exactly this solves the OP's issue.

**Comment by mahrtayyab** [comment]
Replies are only become part of list when both users involved in that reply are member of that list
If user2 is replying to user1 , the reply will only appear in list if user2 is also member of that list

**Comment by harleyLuke** [comment]
Yes sir, you are right. Only users also in the list will get updates for replies， but not all Threads can be obtained in the List, must call Tweet_detail to get all Threads.

**Comment by harleyLuke** [comment]
If you want to get all the replies, I think can only use the database to store the tweets in the list, and then monitor the number of his replies. Maybe you can set up to only monitor the List within 10 days,because not fresh
